iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/SwitchImpl.java

```diff
@@ -85,7 +86,8 @@ public void setRetained(boolean retained) {
         if (oldValue != retained) {
             this.retained.set(index, retained);
             voltageLevel.invalidateCache();
-            network.getListeners().notifyUpdate(this, "retained", oldValue, retained);
+            String variantId = network.getVariantManager().getVariantId(index);
+            network.getListeners().notifyUpdate(this, "retained", variantId, oldValue, retained);
```

Comment:
I think we should notify before invalidating the cache of the voltage level (this will send a lot of notifications).